class.jetpack-jitm.php

```diff
@@ -233,14 +233,10 @@ static function get_messages( $message_path, $query ) {
 
 		// attempt to get from cache
 		$envelopes  = get_transient( 'jetpack_jitm_' . $path );
-		$from_cache = false;
 
-		// if something is in the cache and it was put in the cache after the last heartbeat or sync, use it
-		$last_heartbeat = Jetpack_Options::get_option( 'last_heartbeat', time() );
-		$last_sync      = Jetpack_Options::get_option( 'last_sync', time() );
-		if ( $envelopes && $last_heartbeat < $envelopes['response_time'] || $last_sync < $envelopes['response_time'] ) {
-			$from_cache = true;
-		}
+		// if something is in the cache and it was put in the cache after the last sync we care about, use it
+		$last_sync = (int) get_transient( 'jetpack_last_sync' );
+		$from_cache = $envelopes && $last_sync < $envelopes[ 'response_time' ];
```

Comment:
I can't think of a reason we'd want to cache a jitm that long. It's currently set to 5m by default, which I think is plenty long enough. Most people are in wp-admin to do something, and 5m sounds like a good "gut check refresh" time.

---------

It doesn't *need* to be a transient, it's just simpler that way. No db query required when a caching layer is present. If it expires (or gets evicted, which it's also likely the jitm itself would have been evicted at the same time), they'll get an updated jitm (`0 < any_time()`)